app/user_cleanup.go

```diff
@@ -0,0 +1,94 @@
+package app
+
+import (
+	"strings"
+
+	"github.com/rancher/rancher/pkg/systemaccount"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type userCleanup struct {
+	users    v3.UserInterface
+	clusters v3.ClusterInterface
+	projects v3.ProjectInterface
+}
+
+func cleanupOrphanedSystemUsers(management *config.ManagementContext) {
+	u := userCleanup{
+		users:    management.Management.Users(""),
+		clusters: management.Management.Clusters(""),
+		projects: management.Management.Projects(""),
+	}
+	u.cleanup()
+}
+
+func (u *userCleanup) cleanup() {
+	users, err := u.users.List(v1.ListOptions{})
+	if err != nil {
+		logrus.Errorf("Error listing users during system account users cleanup: %v", err)
```

Comment:
since this happening only once, how would this get cleaned up if just return?

Reply:
oh I had this all in a controller at first and tested it only with all system account users when switched to this one time logic, my bad